plugins/apps/threatbus_misp/threatbus_misp/plugin.py

```diff
@@ -110,11 +117,15 @@ def run(self):
                 continue
             if not is_whitelisted(msg, filter_config):
                 continue
-            intel = map_to_internal(msg["Attribute"], msg.get("action", None), logger)
-            if not intel:
-                logger.debug(f"Discarding unparsable intel {msg['Attribute']}")
-            else:
-                self.inq.put(intel)
+            action = msg.get("action", None)
+            ioc = attribute_to_stix2_indicator(msg["Attribute"], action, logger)
+            if not ioc:
+                if action != "add":
+                    logger.warn(
+                        f"Discarding unparsable MISP attribute {msg['Attribute']}"
+                    )
+                continue
+            self.inq.put(ioc)
```

Comment:
This would be nicer if the function would work like so:
```suggestion
            try:
                ioc = attribute_to_stix2_indicator(msg["Attribute"], action, logger)
            except Exception as e:
                logger.warn(
                     f"Failed to parse MISP attribute {msg['Attribute']}: {e}")
            else:
                self.inq.put(ioc)
```